code/modules/power/lighting.dm

```diff
@@ -340,8 +340,41 @@
 	var/area/A = get_area(src)
 	return A.lightswitch && A.power_light
 
+///Flickers the lights for a set amount of time and flicker delays.
+/obj/machinery/light/proc/set_flicker(amount, flicker_time_lower = 1, flicker_time_upper = 2, flicker_delay = 0.3 SECONDS, ignore_flickering = TRUE)
+	if(status != LIGHT_OK)
+		return
+	if(!has_power())
+		return
+	///Timer id for flickering.
+	var/theflicker
+	if(!ignore_flickering && flickering) //is it flickering when set_flicker-ed
+		//get rid of the stuff to start over so we dont mess stuff up.
+		deltimer(theflicker)
+		resetflickers()
+	random_flicker = TRUE
+	flickering = TRUE
+	if(flicker_time_lower)
+		flicker_time_lower_min = flicker_time_lower
+	if(flicker_time_upper)
+		flicker_time_upper_max = flicker_time_upper
+	flicker(FALSE, FALSE, flicker_delay)
+	theflicker = addtimer(CALLBACK(src, PROC_REF(resetflickers), amount))
+
+///sets values as they were for the light, before set_flicker.
+/obj/machinery/light/proc/resetflickers()
+	flicker_time_lower_min = initial(flicker_time_lower_min)
+	flicker_time_upper_max = initial(flicker_time_upper_max)
+	random_flicker = initial(random_flicker)
+	flickering = FALSE
+	if(!has_power())
+		lightambient.stop(src)
+		return
+	if(status == LIGHT_OK)
+		lightambient.start(src)
+
 ///flicker lights on and off
-/obj/machinery/light/proc/flicker(toggle_flicker = FALSE)
+/obj/machinery/light/proc/flicker(toggle_flicker = FALSE, long_off = TRUE, flicker_delay = 0.3 SECONDS)
```

Comment:
what is long_off? use clear arg names

Reply:
the default behavior where it multiplied the off lights time, i'll note it somewhere.